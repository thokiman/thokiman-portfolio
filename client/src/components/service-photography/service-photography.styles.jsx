import styled from "styled-components";

export const RightServiceContainerPg = styled.div`
  background-color: rgba(18, 129, 148, 0.75);
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
  @media only screen and (max-width: 800px) {
    width: 100%;
    right: 0;
    height: 50vh;
  }
`;

export const RightServiceContentPg = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerPg.displayName = "RightServiceContainerPg";
RightServiceContentPg.displayName = "RightServiceContentPg";
