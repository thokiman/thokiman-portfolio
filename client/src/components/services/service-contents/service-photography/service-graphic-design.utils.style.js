import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../../utils/common.component.utils';

export const measureRightPageContainerPgProps = (viewWidth, viewHeight) => {
  const rightServiceContainerPgHeightOffset = 0;
  const rightServiceContainerPgTopOffset = 0;

  let rightServiceContainerProps;

  if (hasScreenSize1(viewWidth)) {
    console.log('service-photography', 1);
    rightServiceContainerProps = {
      height: 280 + rightServiceContainerPgHeightOffset,
      top: -60 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('service-photography', 2);
    rightServiceContainerProps = {
      height: 270 + rightServiceContainerPgHeightOffset,
      top: -40 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('service-photography', 3);
    rightServiceContainerProps = {
      height: 180 + rightServiceContainerPgHeightOffset,
      top: -50 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('service-photography', 4);
    rightServiceContainerProps = {
      height: 175 + rightServiceContainerPgHeightOffset,
      top: -50 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('service-photography', 5);
    rightServiceContainerProps = {
      height: 220 + rightServiceContainerPgHeightOffset,
      top: -50 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('service-photography', 6);
    rightServiceContainerProps = {
      height: 225 + rightServiceContainerPgHeightOffset,
      top: -50 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('service-photography', 7);
    rightServiceContainerProps = {
      height: 235 + rightServiceContainerPgHeightOffset,
      top: 0 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('service-photography', 8);
    rightServiceContainerProps = {
      height: 235 + rightServiceContainerPgHeightOffset,
      top: -40 + rightServiceContainerPgTopOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('service-photography', 9);
    rightServiceContainerProps = {
      height: 225 + rightServiceContainerPgHeightOffset,
      top: -50 + rightServiceContainerPgTopOffset,
    };
  } else {
    console.log('service-photography', 10);
    rightServiceContainerProps = {
      height: 300,
      top: -50,
    };
  }

  return rightServiceContainerProps;
};
