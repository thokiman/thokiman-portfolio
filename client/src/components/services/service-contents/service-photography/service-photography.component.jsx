import React from 'react';
import { connect } from 'react-redux';
import { selectIsSideBarActive } from 'redux/header/header.selectors';
import { createStructuredSelector } from 'reselect';
import { selectPhotoService } from '../../../../redux/service/service.selectors';
import ServiceIconContent from '../../service-commons/service-icon-content/service-icon-content.component';
import ServiceUrlGeneralContent from '../../service-commons/service-url-general-content/service-url-general-content.component';
import ServiceUrlIconContent from '../../service-commons/service-url-icon-content/service-url-icon-content.component';
import {
  RightServiceContainerPg,
  RightServiceContentPg,
} from './service-photography.styles';

export const ServicePhotography = ({
  photo: { title, description, photoStockItems, photographyIconItems },
  isSideBarActive,
}) => {
  return (
    <RightServiceContainerPg $issidebaractive={isSideBarActive}>
      <RightServiceContentPg>
        <ServiceUrlGeneralContent title={title} description={description} />
        <ServiceIconContent items={photographyIconItems} />
        <ServiceUrlIconContent items={photoStockItems} />
      </RightServiceContentPg>
    </RightServiceContainerPg>
  );
};
const mapStateToProps = createStructuredSelector({
  photo: selectPhotoService,
  isSideBarActive: selectIsSideBarActive,
});
export default connect(mapStateToProps)(ServicePhotography);
