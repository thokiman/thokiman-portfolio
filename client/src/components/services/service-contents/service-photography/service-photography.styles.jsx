import styled, { css } from 'styled-components';

const backgroundColorRightServiceContainerPg = 'rgba(18, 129, 148, 0.75)';
const maxWidthSmartphone = 800;
const lightBlueColor = '#74c6d4';

const ServicePhotographyIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return ServicePhotographyIsNotActivated;
  return null;
};

export const RightServiceContainerPg = styled.div`
  background-color: ${backgroundColorRightServiceContainerPg};
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    overflow-y: hidden;
    width: 100%;
    right: 0;
    height: 280vh;
    ${getSideBarActive}
    @media only screen and (max-width: 550px) {
      height: 270vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 823px) {
      height: 180vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 812px) {
      height: 175vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 736px) {
      height: 220vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 667px) {
      height: 220vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 350px) and (max-height: 568px) {
      height: 235vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 300px) {
      height: 225vh;
      ${getSideBarActive}
    }
  }
`;

export const ServiceStockPhotos = styled.span`
  margin-top: 2vh;
  font-family: futura-demi;
  font-size: 2.5vw;
  margin-bottom: 15px;
  text-indent: 0.6em;
  color: ${lightBlueColor};
  letter-spacing: 0.02em;
  font-kerning: auto;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 6vw;
  }
`;

export const RightServiceContentPg = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerPg.displayName = 'RightServiceContainerPg';
RightServiceContentPg.displayName = 'RightServiceContentPg';
