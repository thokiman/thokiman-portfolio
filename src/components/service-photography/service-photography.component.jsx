import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { CSSTransitionGroup } from "react-transition-group";

import ServiceUrlGeneralContent from "../service-url-general-content/service-url-general-content.component";
import ServiceUrlIconContent from "../service-url-icon-content/service-url-icon-content.component";
import {
  RightServiceContainerPg,
  RightServiceContentPg,
} from "./service-photography.styles";
import { selectPhotoService } from "../../redux/service/service.selectors";

const ServicePhotography = ({ photo: { title, description, items } }) => {
  return (
    <RightServiceContainerPg>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={800}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <RightServiceContentPg>
          <ServiceUrlGeneralContent title={title} description={description} />
          <ServiceUrlIconContent items={items} />
        </RightServiceContentPg>
      </CSSTransitionGroup>
    </RightServiceContainerPg>
  );
};
const mapStateToProps = createStructuredSelector({
  photo: selectPhotoService,
});
export default connect(mapStateToProps)(ServicePhotography);
