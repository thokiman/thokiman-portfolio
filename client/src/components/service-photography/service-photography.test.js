import React from "react";
import { shallow } from "enzyme";

import { ServicePhotography } from "./service-photography.component";

describe("ServicePhotography component", () => {
  let wrapper;
  const mockArray = [];
  const mockText = "";

  beforeEach(() => {
    const mockProps = {
      photo: { title: mockText, description: mockArray, items: mockArray },
    };
    wrapper = shallow(<ServicePhotography {...mockProps} />);
  });

  it("should render ServicePhotography component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
