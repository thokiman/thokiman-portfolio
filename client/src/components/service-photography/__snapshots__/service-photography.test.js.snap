// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicePhotography component should render ServicePhotography component 1`] = `ShallowWrapper {}`;
